platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,236 @@
+        get_rdy_inotify = InotifyHelper(self.get_dpu_rdy_path)
+        dpu_rdy = get_rdy_inotify.add_watch(WAIT_FOR_DPU_READY, 1)
+        if not dpu_rdy:
+            if forced:
+                if count > 1:
+                    time.sleep(1)
+                    self.dpu_power_off(forced=True)
+                    self.dpu_power_on(forced=True, count=count-1)
+                else:
+                    print(f"{self.get_name()}: Failed Force power on! Exiting")
+                    return False
+            else:
+                self.dpu_power_off(forced=True)
+                self.dpu_power_on(forced=True)
+        else:
+            print(f"{self.get_name()}: Power on Successful!")
+            return True
+
+    def dpu_reboot_prep(self):
+        """Per DPU Reboot API"""
+        # TODO: Shutdown SONiC on DPU -> SSH Connection -> Shutdown script
+
+    def dpu_burn_fw(self, path):
+        """Per DPU Firmware Update API"""
+        # TODO: Uncomment to install the bfb Image
+        """if not os.path.isfile(path):
+            raise FileNotFoundError(f"{self.get_name()}:File "
+                                    f"{self.file_name} does not exist!")
+        cmd = ["sonic_bfb_install" ,"-b" ,path,"-r","rshim"+str(self.index)]
+        try:
+            cmd_output = subprocess.check_output(cmd)
+        except subprocess.CalledProcessError as cmd_exc:
+            print("Installation failed! code",
+                  cmd_exc.returncode,
+                  cmd_exc.output)"""
+
+    def dpu_pci_scan(self):
+        """PCI Scan API"""
+        set_pci_scan = "/sys/bus/pci/rescan"
+        self.write_file(set_pci_scan, "1")
+
+    def dpu_pci_remove(self):
+        """Per DPU PCI remove API"""
+        get_dpu_pci_path = os.path.join(CONFIG_BASE,
+                                        f"dpu{self.index}_pci_bus_id")
+        pci_string = utils.read_str_from_file(get_dpu_pci_path,
+                                              raise_exception=True)
+        get_pci_dev_path = "/sys/bus/pci/devices/"+pci_string+"/remove"
+        self.write_file(get_pci_dev_path, "1")
+
+    def dpu_fw_upgrade(self, path):
+        """Per DPU Firmware Upgrade API"""
+        print(f"{self.get_name()}: FW upgrade")
+        self.dpu_burn_fw(path)
+        self.dpu_reboot_prep()
+        self.dpu_pci_remove()
+        self.write_file(self.set_dpu_perst_en_path, "0")
+        self.dpu_go_down()
+        self.write_file(self.set_dpu_perst_en_path, "1")
+        get_rdy_inotify = InotifyHelper(self.get_dpu_rdy_path)
+        dpu_rdy = get_rdy_inotify.add_watch(WAIT_FOR_DPU_READY, 1)
+        if not dpu_rdy:
+            self.dpu_power_off(forced=True)
+            self.dpu_power_on(forced=True)
+        self.dpu_pci_scan()
+        print(f"{self.get_name()}: FW upgrade complete")
+
+    def dpu_reboot(self):
+        """Per DPU Reboot API"""
+        print(f"{self.get_name()}: Reboot")
+        self.dpu_reboot_prep()
```

Comment:
dpu_reboot_prep
dpu_pci_remove
dpu_pci_rescan
are not needed

Reply:
Done